### ISSUE TITLE
Replace Flash With HTML5

### ISSUE THREAD

**Issue by Elshara** [SocialEngine/phpv4-issues#686]
(Thanks for reporting an issue! Please make sure you click the link above to view the issue guidelines, then fill out the blanks below.)

What are the steps to reproduce this issue?
-------------------------------------------
1. … Go to the uploaders section of any page. Add photo, music, video, or package. Also you can find the same by going to the file manager in media file manager in the admin panel.
2. … Notice the flash uploader still exists.
3. … As was documented in the recent changes of version 4.9, there is no ability to change the upload type from html5 to flash and back again.

What happens?
-------------
… The idea that multiple uploaders exist in social engine, began to surface during the time the change log became available prior to the 4.9 release being live for everyone on April 12. The issue is that there still doesn't seem to be a way to actually choose the type of uploader present when uploading any file to the site what so ever. Mobile browsers do not support flash and it is removed in both the google and apple stores where all mobile applications are housed. Leaving, by default, only desktop laptop markets to surface flash deprecated flash content. This has to change. If flash content isn't available on mobile, then this means half of your website visitors at least, if not most of them, cannot upload any file type. The uploder used when originally joining a social engine website is actually the only html5 based uploader present on the site itself. This uploader should be standard as it is only used right now, to upload profile photos. This should likewise be used to upload all other file types as well. effectively, bringing flash era to the past. I also believe the chat sound uses flash, and if you disable it to be replaced with an html5 control, you will actually have a much faster site with chat enabled by default, the default friend live personal messenger being what it is  in the chat module itself.

What were you expecting to happen?
----------------------------------
… Clear explanation as to how to make html5 uploading possible.

Any logs, error output, etc?
----------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.) 

```
Paste Log Here
```

Any other comments?
-------------------
… If I'm mistaken, I apologize now in advance. I'm convinced however, this is not a feature yet in social engine.

What versions of software are you using?
----------------------------------------
**Operating System:** … Windows 10 personal. Linux web server.

**SocialEngine PHP Version:** … 4.9.0

**Comment by DonnaScriptTechs** [comment]
We have a fallback for when flash is not detected. I will go test with my iPhone on our demo to make sure it's working.

**Comment by DonnaScriptTechs** [comment]
Our photo size limits restrict it to small sizes and my phone only has large images. I will have to test on local.

**Comment by DonnaScriptTechs** [comment]
I was able to find a smaller photo. Here it is, uploaded from iPhone 5c, https://demo.socialengine.com/albums/photos/view/21/1191

**Comment by nizhnyk** [comment]
It's just a basic FileUpload input fallback, you can't upload multiple photos with it and there is no status while you're uploading photos. What we want is a full replacement of FancyUpload, multiple file input with nice real-time status of files being uploaded, all using just HTLM5.

**Comment by DonnaScriptTechs** [comment]
https://github.com/SocialEngine/phpv4-issues/issues/129 This is a duplicate of that request. Which one do you want to remain open @Elshara ?

**Comment by DonnaScriptTechs** [comment]
I closed the other one since this one has clients in it already and both were for the same thing.

**Comment by SpiritWolfie** [comment]
I agree with this in full!

Flash by default is blocked by most browsers, most websites don't even support flash anymore nor do most platforms!, and allot of peoples computers can't use flash correctly for god knows what reasons

Taken directly from chromes official faq
https://support.google.com/chrome/answer/6258784?hl=en

you have to 'turn on' flash
Step 1: Turn on Adobe Flash Player

then it also clearly states toward the bottom 
Info for administrators
Critical vulnerabilities are regularly found in Adobe Flash Player, and are quickly used in exploit kits.

It also states that chrome will also block flash content if it 'decides' something isn't safe, so even if you have it turned on that wont mean a dang thing..

Firefox if I remember correctly has flash hard blocked

Html 5 needs to be used for anything in regards to 'uploading' so people will stop having issues with getting the photo upload working etc... We don't need this in SE 5.0 we needed this in SE 4.9 or 4.9. whatever... Flash should have been yanked out

If you can not see this then I just don't know what else to say....

**Comment by Elshara** [comment]
Keep this issue open, we'll use it.
Generally if when browsers deprecate something, it's for a good
reason. Code browsers use to communicate between computers and web
servers, should generally follow a similar practice by design,
especially paid platforms.

**Comment by gsf00001** [comment]
Amen! Get rid of Flash and use HTML5 (or whatever is best/current at the time this occurs).

**Comment by Elshara** [comment]
My Google Pixil won't even support flash. the operating system is
version 7 and the last version of flash was made for version 4 of
Android. Similarly, Iphone users I notice have trouble with it also.
No telling whether or not tablets and ipads are the same.

**Comment by DonnaScriptTechs** [comment]
We've had plans to look at this for future versions. I've added it to our list to discuss. We already had it in our list to discuss for flowplayer.

**Comment by Elshara** [comment]
Okay cool. There's always say JPlayer, JWPlayer, Kaltura among others.
To give you a few examples.
https://www.similartech.com/compare/flowplayer-vs-jplayer
http://www.streamingmedia.com/Articles/Editorial/Featured-Articles/Choosing-a-Video-Player-Features-and-Specs-for-the-Top-Five-94188.aspx
Food for thought. I think Boonex uses JPlayer.

**Comment by SpiritWolfie** [comment]
Flowplayer is just a video HTML 5.. from what I'm reading?

we need flash gone for photo uploads primary, as allot of people are having issues uploading photos on older machines, and the fact browsers are starting to hard block flash these days.

**Comment by DonnaScriptTechs** [comment]
@SpiritWolfie that was a mistype on my part. I meant flash and not flowplayer. We were talking about flash and for some reason I typed flowplayer. Sorry for the confusion.

**Comment by SpiritWolfie** [comment]
@DonnaScriptTechs 

That's fine, I was just confused, but I agree that we need this sooner then later, I have countless people on my website complaining they can't upload photos on mobile or none mobile... and on rare occasions even I've had issues with this on fresh installs.

the fall back doesn't seem to work correctly, but we shouldn't need to fall back to a none multi select ability's ether, as many other platforms have pretty much zero issues with multi uploads and photo uploads on any device.

Thanks for clarifying.

**Comment by Elshara** [comment]
I only hope it will be easily implemented. Otherwise, replacing the
form will have to be done by manual override. That's not fun but I'll
do it if I have to.

**Comment by GarfieldS** [comment]
Wow! In terms of language is this really something that needs to be discussed or debate? You may as well say building a responsive web site is something we are looking into and need to discuss. If videos are not flash responsive means nothing to a site who uses video as a key aspect of running their network. 

If what is being said here is true 4.9 is already outdated compared to Ning 3.0/Jamroom whose user base you are asking to migrate over to currently an outdated video solution. Given the amount of time it has taken for 4.9 to come out. This statement "We've had plans to look at this for future versions. I've added it to our list to discuss." is not encouraging.

**Comment by DonnaScriptTechs** [comment]
@GarfieldS when we say "we've added it to our list to discuss" this is the standard phrase we use for items we've added to our internal feature list. I am not a core developer. I add things to the list for improvements. As this was added, it is something we will be improving.

Video already uses html5 and has had the option for a while. 

<img width="697" alt="videohtml5" src="https://cloud.githubusercontent.com/assets/6908282/25479631/99da9bb4-2b13-11e7-9679-9a6c6f4e8908.png">

This request was for other areas that do not use it.

**Comment by GarfieldS** [comment]
Ok, apologies about that. I didn't get that distinction. Thank you for pointing this out in this discussion and not just for my sake. When flash and html 5 is mentioned video immediately comes to mind.

**Comment by Elshara** [comment]
To be honest, the only place flash is called are uploaders and chat.

**Comment by SpiritWolfie** [comment]
@Elshara 

Their photo upload uses flash too....from what i've seen, and it relays very heavy on it

**Comment by Elshara** [comment]
I know. So does music and videos. Packages two.

**Comment by DonnaScriptTechs** [comment]
This issue was moved to SocialEngine/phpv4-feature-requests#100

**Comment by DonnaScriptTechs** [comment]
In our efforts to streamline our processes for bug fixes and feature implementation, we are separating our issue tracker and feature request section. This has been moved to our feature area where you are subscribed automatically in order to get updates. Thank you for your contribution.
